packages/lib/default-cookies.ts

```diff
@@ -0,0 +1,72 @@
+import { CookiesOptions } from "next-auth";
+
+import { isENVDev } from "@calcom/lib/env";
+
+/**
+ * Copy from 'https://github.com/nextauthjs/next-auth/blob/227ff2259f/src/core/lib/cookie.ts' as we can't import it directly
+ *
+ * Use secure cookies if the site uses HTTPS
+ * This being conditional allows cookies to work non-HTTPS development URLs
+ * Honour secure cookie option, which sets 'secure' and also adds '__Secure-'
+ * prefix, but enable them by default if the site URL is HTTPS; but not for
+ * non-HTTPS URLs like http://localhost which are used in development).
+ * For more on prefixes see https://googlechrome.github.io/samples/cookie-prefixes/
+ *
+ */
+
+export function defaultCookies(useSecureCookies: boolean): CookiesOptions {
+  const cookiePrefix = useSecureCookies ? "__Secure-" : "";
+  // To enable cookies on widgets,
+  // https://stackoverflow.com/questions/45094712/iframe-not-reading-cookies-in-chrome
+  // But we need to set it as `lax` in development
+  const sameSite = isENVDev ? "lax" : "none";
```

Comment:
This was working, but we need to make sure we're using securized cookies on prod, I use this isENVDev to detect the envirorment, also defaultCookies accepts a boolean to useSecureCookies or not. passing the current URL to it with startsWith(https://...) does the trick